v1/api/auth.ts

```diff
@@ -0,0 +1,56 @@
+import { Type, Static } from '@sinclair/typebox'
+import { CAPABILITIES_ARRAY, CAPABILITIES, getExpiry, subset } from '../authorization/jwt.js'
+import { StoreI } from '../config/index.js'
+import { FastifyTypebox } from './index.js'
+
+export const authRoutes = (store: StoreI) => async (server: FastifyTypebox): Promise<void> => {
+  server.post<{
+    Body: Static<typeof CAPABILITIES_ARRAY>
+  }>('/auth/exchange', {
+    schema: {
+      body: CAPABILITIES_ARRAY,
+      response: {
+        200: Type.String(),
+        401: Type.String(),
+      },
+      description: 'Exchange an old token for a new one with a subset of initial capabilities and an updated expiry time. Takes in the old token from the Authorization header',
+      tags: ['admin', 'publisher'],
+      security: [{ jwt: [] }]
+    },
+    preHandler: server.auth([server.verifyRefresh])
+  }, async (request, reply) => {
+    const token = request.user
+    if (!subset(request.body, token.capabilities)) {
+      return reply.status(401).send("Requested more permissions than original token has")
+    }
+    if (!token.capabilities.includes(CAPABILITIES.ADMIN)) {
```

Comment:
Refresh tokens don't expire atm (general consensus seems to either have these have really long expiry or not expire at all)

Reply:
That makes sense.

Was my assumption that this prevents non-admins from using the endpoint accurate?

Reply:
Yes, that's correct

Reply:
Was there another endpoint that publishers should be using to refresh their token or create new ones? IMO it'd be useful to just use the one.

Reply:
Oh sorry I meant that publishers can't make new refresh tokens but they can still use this endpoint to exchange their existing token for a new one with a refreshed expiry date.

Reply:
Oh ok cool. I was just confused by the `if (!token.capabilities.includes(CAPABILITIES.ADMIN)) throw an error`. Should that be within the if statement above instead? Like "if it isn't a subset, and it isn't an admin token, throw an error" instead of "if it isn't a subset, throw an error; if it isn't an admin throw an error"

Reply:
going to replace it with `!token.capabilities.includes(CAPABILITIES.ADMIN) && request.body.includes(CAPABILITIES.REFRESH)`